DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h

```diff
@@ -120,7 +124,12 @@ class SiPixelRecHitQuality {
       qualWord |= ((raw & probX_mask) << probX_shift);
     }
     inline void setProbabilityQ(float prob, QualWordType& qualWord) const {
-      if (prob < 0 || prob > 1) {
+      if (prob < 0.0f) {
+        warningOutOfBoundProb("Q", prob, qualWord);
+        prob = 0;
+      } else if (prob > 1.0f && prob < 1.0f + std::numeric_limits<float>::epsilon()) {
+        prob = 1;
+      } else if (prob > 1.0f + std::numeric_limits<float>::epsilon()) {
```

Comment:
```suggestion
      if (prob > 1.0f && prob < 1.0f + std::numeric_limits<float>::epsilon()) {
        prob = 1;
      } else if (prob < 0.0f || prob > 1.0f + std::numeric_limits<float>::epsilon()) {
```
is more compact and also puts this to me to within a threshold for not refactoring this range check here and in `setProbabilityXY` to a separate method.

Reply:
(the suggestion was edited) we should also have `prob > 1.0f && prob <= 1.0f + std::numeric_limits<float>::epsilon()`
otherwise, `prob > 1.0f && prob < 1.0f + std::numeric_limits<float>::epsilon()` is a null set.